src/BeamAdapter/component/BeamInterpolation.inl

```diff
@@ -18,1522 +18,1550 @@
+    d_dofsAndBeamsAligned.setValue(false);
+    DOF0TransformNode0.push_back(Transform(Vec3(0, 0, 0), QuatX ));
+    DOF1TransformNode1.push_back(Transform(Vec3(0, 0, 0), QuatX ));
+
+    d_edgeList.endEdit();
+    d_lengthList.endEdit();
+    d_DOF0TransformNode0.endEdit();
+    d_DOF1TransformNode1.endEdit();
+    d_curvAbsList.endEdit();
+}
+
+
+template<class DataTypes>
+void BeamInterpolation<DataTypes>::getBeamAtCurvAbs(const Real& x_input, unsigned int &edgeInList_output, Real& baryCoord_output, unsigned int start)
+{
+    /// lTotalRest = total length of the
+    Real lTotalRest = getRestTotalLength();
+    /// LTotal =  length sum of the beams that are "out"
+    Real LTotal=0.0;
+
+    const unsigned int edgeListSize = d_edgeList.getValue().size();
+
+    /// we find the length of the beam that is "out"
+    for (unsigned int e = start; e < edgeListSize; e++)
+    {
+        LTotal += getLength(e);
+    }
+
+    /// x_i = abs_curv from the begining of the instrument
+    Real  x_i = x_input + LTotal - lTotalRest;
+
+    if( x_i < 0.0)
+    {
+        edgeInList_output = start;
+        baryCoord_output = 0;
+        return;
+    }
+
+    /// we compute the x value of each node :the topology (stored in Edge_list) is supposed to be a regular seq of segment
+    Real x = 0;
+
+    for (unsigned int e = start; e < edgeListSize; e++)
+    {
+        x += getLength(e);
+        if(x > x_i)
+        {
+            edgeInList_output = e;
+            Real x0 = x - getLength(e);
+            baryCoord_output =(x_i-x0) / getLength(e);
+            return;
+        }
+    }
+
+    edgeInList_output = edgeListSize - 1;
+    baryCoord_output = 1.0;
+}
+
+
+template <class DataTypes>
+void BeamInterpolation<DataTypes>::getSamplingParameters(type::vector<Real>& /*xP_noticeable*/, type::vector< int>& /*nbP_density*/)
+{
+    msg_error()<<"getSamplingParameters is not implemented when _restShape== nullptr : TODO !! ";
+}
+
+template <class DataTypes>
+typename BeamInterpolation<DataTypes>::Real BeamInterpolation<DataTypes>::getRestTotalLength()
+{
+    Real le(0.0);
+    const vector< double > &lengthList = d_lengthList.getValue();
+
+    for (unsigned int i = 0; i < lengthList.size(); i++)
+        le += lengthList[i];
+
+    return le;
+}
+
+template <class DataTypes>
+void BeamInterpolation<DataTypes>::getCollisionSampling(Real &dx, const Real& /*x_localcurv_abs*/)
+{
+    unsigned int numLines = 30;
+    dx = getRestTotalLength()/numLines;
+}
+
+template <class DataTypes>
+void BeamInterpolation<DataTypes>::getNumberOfCollisionSegment(Real &dx, unsigned int &numLines)
+{
+    numLines = 30;
+    dx = getRestTotalLength()/numLines;
+}
+
+template <class DataTypes>
+void BeamInterpolation<DataTypes>::getYoungModulusAtX(int beamId,Real& /*x_curv*/, Real& youngModulus, Real& cPoisson)
+{
+    youngModulus = Real(d_defaultYoungModulus.getValue()[beamId]);
+    cPoisson     = Real(d_poissonRatio.getValue()[beamId]);
+}
```

Comment:
My changes.

Reply:
This is a problem I see a lot in this plugin 🥸 but you should check if the beamId is within the bounds